hedera-node/test-clients/src/main/java/com/hedera/services/bdd/junit/TokenReconciliationValidator.java

```diff
@@ -0,0 +1,80 @@
+/*
+ * Copyright (C) 2023 Hedera Hashgraph, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hedera.services.bdd.junit;
+
+import static com.hedera.services.bdd.junit.TestBase.concurrentExecutionOf;
+
+import com.hedera.services.bdd.junit.utils.AccountClassifier;
+import com.hedera.services.bdd.junit.validators.AccountNumTokenNum;
+import com.hedera.services.bdd.suites.records.TokenBalanceValidation;
+import com.hedera.services.stream.proto.RecordStreamItem;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Stream;
+
+/**
+ * This validator "reconciles" the HTS token balances of all accounts between the record
+ * stream and the network state, comparing two sources of truth at the end of the CI test run:
+ *
+ * <ol>
+ *   <li>The balances implied by the {@code TransferList} adjustments in the record stream.
+ *   <li>The balances returned by {@code getTokenAccountBalance} queries.
+ * </ol>
+ *
+ * <p>It uses the {@link com.hedera.services.bdd.suites.records.TokenBalanceValidation} suite to perform the queries.
+ */
+public class TokenReconciliationValidator implements RecordStreamValidator {
+    private final Map<AccountNumTokenNum, Long> expectedTokenBalances = new HashMap<>();
+
+    private final AccountClassifier accountClassifier = new AccountClassifier();
+
+    @Override
+    @SuppressWarnings("java:S106")
+    public void validateRecordsAndSidecars(final List<RecordWithSidecars> recordsWithSidecars) {
+        getExpectedBalanceFrom(recordsWithSidecars);
+        System.out.println("Expected token balances: " + expectedTokenBalances);
```

Comment:
I switched to using a logger. The System.out was there because I had cribbed from BalanceReconciliationValidator, but there's no reason not to use a logger.